src/symbol/shaping.js

```diff
@@ -366,18 +372,19 @@ function determineLineBreaks(logicalInput: TaggedString,
 
         // Ideographic characters, spaces, and word-breaking punctuation that often appear without
         // surrounding spaces.
-        if ((i < logicalInput.length() - 1) &&
-            (breakable[codePoint] ||
-                charAllowsIdeographicBreaking(codePoint))) {
-
-            potentialLineBreaks.push(
-                evaluateBreak(
-                    i + 1,
-                    currentX,
-                    targetWidth,
-                    potentialLineBreaks,
-                    calculatePenalty(codePoint, logicalInput.getCharCode(i + 1)),
-                    false));
+        if ((i < logicalInput.length() - 1)) {
+            const ideographicBreak = !breakable[codePoint] && charAllowsIdeographicBreaking(codePoint);
```

Comment:
I dont see any overlap between `charAllowsIdeographicBreaking`and `breakable` codepoints. This check could be just  `charAllowsIdeographicBreaking`;